Mathlib/Data/List/Range.lean

```diff
@@ -234,4 +234,93 @@ theorem nthLe_finRange {n : ℕ} {i : ℕ} (h) :
   have h₂ : (finRange k).get ⟨i, by simp⟩ = i := get_finRange _
   simpa using (Nodup.get_inj_iff (nodup_finRange k)).mp (Eq.trans h₁ h₂.symm)
 
+section Ranges
+
+
+/-- From `l: List ℕ`, construct `ml: List (List ℕ)` such that
+  `ml.map List.length = l` and `ml.join = range l.sum` -/
```

Comment:
```suggestion
/-- From `l : List ℕ`, construct `l.ranges : List (List ℕ)` such that
  `ml.map List.length = l` and `ml.join = range l.sum` -/
```
Can you maybe add a short example? I find a little difficult to read the definition.

Reply:
I just did that as requested : [1,2,3].range = [[0],[1,2],[3,4,5]. I also tstarted proving the converse of the last statement (ranges_lt) which, together with the preceding results characterizes `ranges`.